common/utils.go

```diff
@@ -90,29 +90,24 @@ func linkToNetDev(link netlink.Link) (NetDev, error) {
 	return netDev, nil
 }
 
-// Lookup the weave interface of a container
-func GetWeaveNetDevs(processID int) ([]NetDev, error) {
-	// Bail out if this container is running in the root namespace
-	nsToplevel, err := netns.GetFromPid(1)
-	if err != nil {
-		return nil, fmt.Errorf("unable to open root namespace: %s", err)
-	}
-	nsContainr, err := netns.GetFromPid(processID)
-	if err != nil {
-		return nil, fmt.Errorf("unable to open process %d namespace: %s", processID, err)
-	}
-	if nsToplevel.Equal(nsContainr) {
-		return nil, nil
-	}
+// ConnectedToBridgePredicate returns a function which is used to query whether
+// a given link is a veth interface which one end is connected to a bridge.
+// The returned function should be called from a container network namespace which
+// the bridge does NOT belong to.
+func ConnectedToBridgePredicate(bridgeName string) (func(link netlink.Link) bool, error) {
+	var br netlink.Link
 
-	weaveBridge, err := netlink.LinkByName("weave")
+	err := weavenet.WithNetNSLinkByPid(1, bridgeName, func(link netlink.Link) error {
```

Comment:
Good spot. Anyway, I'm wrapping both with `WithNetNS*`, because the `GetWeaveNetDevs`, which calls the predicate function, is used by the proxy and atm `WithNetNS*` is not safe.